### PR TITLE
Community channel editing

### LINKED ISSUE #12065: Add description to community channels
As an admin
I want to be able to add a description to a community channel
So I can explain to the community members what this channel is about

### Description
A new field should be added when creating/editing a community channel with a description.
I believe the backend already supports it, but needs to be checked.

### Designs

https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=16052%3A2354

### PR DESCRIPTION
**Pull request by vkjr**
fixes #12065

### Summary

- channel details page added
- channel description can be added on creation 
- channel name and description can be edited

[Related status-go PR](https://github.com/status-im/status-go/pull/2236)

#### Platforms
- Android
- iOS

##### Functional

Community channels editing

### Steps to test
- Open Status
- Create a community or open one where you are an admin
- Create a new channel and make sure you can enter description
- Open channel profile, select edit button (top right) and change name/description
- Make sure changes are seen in a channels list
- 
status: ready

### CODE REVIEW

**Comment by churik** [comment]
@vkjr 
e2e failures are not related to your PR
### ISSUE 1: invited user has option to edit chat name / descritpion

**Steps:**
1) User A: Create community ("Require approval") with 1 public chat with description
2) User A: invite user B
3) User B: join community, try to edit community channel

**Expected result:**
edit option is not available
**Actual result:**
edit option is available, when trying to save nothing happened

**OS:** IOS, Android

### ISSUE 2: channel description changes are not saved

**Steps:**
1) User A: Create community ("Require approval") with 1 public chat with description
2) User A: invite user B
3) User B: join community
4) User A: change community description

**Expected result:**
User B: can see that community description is changed
**Actual result:**
User B: nothing changed

**OS:** IOS, Android

### ISSUE 3: long tap > open details from community screen leads to empty details

**Steps:**
1) User A: Create community ("Require approval") with 1 public chat with description
4) User A: open community view with channel list, long tap on created channel > View details
IMAGE 2021-06-01 12:25:48

**OS:** IOS, Android

**Comment by vkjr** [comment]
@churik, all issues were addressed (thanks for finding them)

File: src/status_im/communities/core.cljs [hunk 1]
```diff
@@ -292,10 +317,22 @@
 
 (fx/defn create-channel-pressed
   {:events [::create-channel-pressed]}
-  [cofx id]
+  [{:keys [db] :as cofx} id]
+  (log/info "")
```
Comment (shivekkhurana) [inline src/status_im/communities/core.cljs]: Rogue log.

File: src/status_im/communities/core.cljs [hunk 2]
```diff
@@ -292,10 +317,22 @@
...
             (reset-community-id-input id)
-            (navigation/navigate-to :create-community-channel nil)))
+            (reset-channel-info)
+            (navigation/navigate-to :communities {:screen :create-community-channel})))
+
+(fx/defn edit-channel-pressed
+  {:events [::edit-channel-pressed]}
+  [{:keys [db] :as cofx} community-id chat-name description color]
+  (fx/merge cofx
+            {:db (assoc db :communities/create-channel {:name        chat-name
```
Comment (shivekkhurana) [inline src/status_im/communities/core.cljs]: Can align map k/v pairs.

File: src/status_im/ui/screens/chat/sheets.cljs [hunk 3]
```diff
@@ -86,7 +86,11 @@
       {:theme    :accent
        :title    chat-name
        :icon     [chat-icon/chat-icon-view-chat-sheet
-                  chat-id group-chat chat-name color]}]
+                  chat-id group-chat chat-name color]
```
Comment (shivekkhurana) [inline src/status_im/ui/screens/chat/sheets.cljs]: Map kvs not aligned.

File: src/status_im/ui/screens/communities/channel_details.cljs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                                                                                         community-id
+                                                                                         chat-name
+                                                                                         description
+                                                                                         color])}])
+                           :extended-header   (profile-header/extended-header
+                                               {:title    chat-name
+                                                :color    color
+                                                :subtitle (i18n/label :t/public-channel)})
+                           :use-insets        true}
+      [:<>
```
Comment (shivekkhurana) [inline src/status_im/ui/screens/communities/channel_details.cljs]: Redundant fragment `[:<>`

**Review by shivekkhurana — CHANGES_REQUESTED** [review]
Looks good, pointed out some cosmetic issues.

**Comment by vkjr** [comment]
@shivekkhurana, thanks for the notes, addressed them.

**Review by shivekkhurana — APPROVED** [review]

**Comment by qoqobolo** [comment]
Checked the fixes for issues 1-5, everything works fine now. Thank you @vkjr!
PR can be merged.

**Comment by vkjr** [comment]
@qoqobolo, thanks!

**Pull request merged**